lib/ta_volume.js

```diff
@@ -0,0 +1,62 @@
+// allows to use all talib Volume Indicator Functions: https://mrjbq7.github.io/ta-lib/func_groups/volume_indicators.html
+// AD - Chaikin A/D Line
+// ADOSC - Chaikin A/D Oscillator
+// OBV - On Balance Volume
+
+var talib = require('talib')
+
+module.exports = function volume(s, min_periods, indicator, fastperiod, slowperiod) {
+  return new Promise(function(resolve, reject) {
+    // create object for talib. only close is used for now but rest might come in handy
+    if (!s.marketData) {
+      s.marketData = { open: [], close: [], high: [], low: [], volume: [] }
+    }
+
+    if (s.lookback.length > s.marketData.close.length) {
+      for (var i = (s.lookback.length - s.marketData.close.length) - 1; i >= 0; i--) {
+        s.marketData.high.push(s.lookback[i].high)
+        s.marketData.low.push(s.lookback[i].low)
+        s.marketData.close.push(s.lookback[i].close)
+        s.marketData.volume.push(s.lookback[i].volume)
+      }
+    }
+
+    if (s.marketData.close.length < min_periods) {
+      resolve()
+      return
+    }
+
+    let tmpHigh = JSON.parse(JSON.stringify(s.marketData.high))
```

Comment:
The JSON stringify/parse is an incredibly slow way to make an array copy. Please consider `array.slice(0)` instead. [array.slice() documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) Benchmark: http://jsben.ch/dzO4U

Reply:
thx for this input done